shared/middleware/helpers/assets.rb

```diff
@@ -0,0 +1,68 @@
+#
+# Assets
+#
+class AssetBucket
+
+  ALLOWED_FILE_TYPES = %w(jpg jpeg gif png mp3)
+
+  def initialize
+    params = {region: 'us-east-1'}
+    if CDO.s3_access_key_id && CDO.s3_secret_access_key
+      params[:credentials] = Aws::Credentials.new(CDO.s3_access_key_id, CDO.s3_secret_access_key)
+    end
+    @s3 = Aws::S3::Client.new(params)
+  end
+
+  def list(encrypted_channel_id)
+    owner_id, channel_id = storage_decrypt_channel_id(encrypted_channel_id)
+    prefix = "#{CDO.assets_s3_directory}/#{owner_id}/#{channel_id}"
```

Comment:
Do we have a helper for constructing this path (e.g. s3_path(owner_id, channel_id)).

This would be easier to read and protect us against changes in the path format.